DataStructures/BitList.cs

```diff
@@ -0,0 +1,1750 @@
+    public class ConversionErrorException: Exception { }
+
+    /// <summary>
+    /// A list of bits.
+    /// </summary>
+    /// <exception cref="EmptyInstanceException" />
+    /// <exception cref="BooleanAlgebraException" />
+    /// <exception cref="FileReadingException" />
+    /// <exception cref="FileWritingException" />
+    /// <exception cref="ConversionErrorException" />
+    public sealed class BitList:
+        IList<bool>,
+        IEquatable<BitList>
+    {
+        private BitArray bits;
+
+        /// <summary>
+        /// No params ctor.
+        /// </summary>
+        public BitList()
+        {
+            bits = null;
+        }
+
+        /// <summary>
+        /// Ctor (Copy the <paramref name="bits"/> <see cref="BitList"/> object).
+        /// </summary>
+        public BitList(BitList bits)
+        {
+            this.bits = new BitArray(bits.bits);
+        }
+
+        /// <summary>
+        /// Ctor (<paramref name="unmanagedNonPrimitiveBits"/> is an unmanaged value type object).
+        /// </summary>
+        public BitList(ValueType unmanagedNonPrimitiveBits)
+        {
+            try
+            {
+                if (unmanagedNonPrimitiveBits.GetType().IsPrimitive)
+                {
+                    throw new ConversionErrorException();
+                }
+
+                //Throw an exception if the type of unmanagedBits isn't unmanaged; otherwise, continue.
+                typeof(Unmanaged<>).MakeGenericType(unmanagedNonPrimitiveBits.GetType());
+
+                //Unit of measure: byte (8 bit).
+                var size = Marshal.SizeOf(
+                    unmanagedNonPrimitiveBits.GetType().IsEnum ?
+                        Enum.GetUnderlyingType(unmanagedNonPrimitiveBits.GetType()) : unmanagedNonPrimitiveBits.GetType()
+                );
+
+                if (unmanagedNonPrimitiveBits.GetType().IsEnum)
+                {
+                    switch (Enum.GetUnderlyingType(unmanagedNonPrimitiveBits.GetType()).Name)
+                    {
+                        case "Byte":
+                            bits = new BitArray(new[] { (byte)unmanagedNonPrimitiveBits });
+                            break;
+
+                        case "SByte":
+                            bits = new BitArray(new[] { (byte)(sbyte)unmanagedNonPrimitiveBits });
+                            break;
+
+                        case "Int16":
+                            bits = new BitArray(BitConverter.GetBytes((short)unmanagedNonPrimitiveBits));
+                            break;
+
+                        case "UInt16":
+                            bits = new BitArray(BitConverter.GetBytes((ushort)unmanagedNonPrimitiveBits));
+                            break;
+
+                        case "Int32":
+                            bits = new BitArray(new[] { (int)unmanagedNonPrimitiveBits });
+                            break;
+
+                        default:
+                        case "UInt32":
```

Comment:
Codacy found an issue: [Remove this empty 'case' clause.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=5559704)